feat: gcp secret manger support for DVT

mokhahmed opened the pull request:
/gcbrun

mokhahmed committed:
adding support for secret manger [GCP]

mokhahmed committed:
adding support for secret manger [GCP]

mokhahmed committed:
adding support for secret manger [GCP]

mokhahmed committed:
adding support for secret manger [GCP]

mokhahmed committed:
Revert "adding support for secret manger [GCP]"

This reverts commit b959cdaf5b4f7953e11cd7a9abef60329dc48399.

mokhahmed committed:
adding support for secret manger [GCP]

mokhahmed committed:
Merge remote-tracking branch 'origin/develop' into secret-manager

# Conflicts:
#	docs/connections.md

mokhahmed committed:
Merge remote-tracking branch 'origin/develop' into secret-manager

# Conflicts:
#	docs/connections.md

mokhahmed committed:
Merge remote-tracking branch 'origin/develop' into secret-manager

# Conflicts:
#	docs/connections.md

nehanene15 commented on data_validation/cli_tools.py:
```diff
@@ -368,16 +368,28 @@ def _configure_validation_config_parser(subparsers):
+        "--secret-manger-project-id",
+        "-sm-prj-id",
+        default=None,
+        help="Secret manger type to store credentials by default will be plain ",
```
This description is for type rather than project-id

nehanene15 commented on data_validation/secret_manger.py:
```diff
@@ -0,0 +1,53 @@
+
+    def __init__(self):
+        # Import the Secret Manager client library.
+        from google.cloud import secretmanager
```
Do we need to add google-cloud-secret-manager in our requirements?

mokhahmed replied:
added the dependencies to setup.py

nehanene15 commented on data_validation/secret_manger.py:
```diff
@@ -0,0 +1,53 @@
+class SecretMangerBuilder:
```
Let's add the Google copyright at the top of this - you can just copy/paste from the other files

nj1973 replied:
Also the file is named "secret_manger.py" rather than "secret_manager.py".

mokhahmed replied:
adding copyright headers

nehanene15 commented on data_validation/secret_manger.py:
```diff
@@ -0,0 +1,53 @@
+
+    os.environ[
+        "GOOGLE_APPLICATION_CREDENTIALS"
+    ] = "/Users/moukhtar/ws/ma-sabre-sandbox-01-dfe8f33bb3ad.json"
```
We should redact this path and the secret on line 52

mokhahmed replied:
removed

nehanene15 commented on docs/connections.md:
```diff
@@ -55,11 +60,11 @@ The data validation tool supports the following connection types.
 * [AlloyDB](#AlloyDB)
 
-As you see above, Teradata and BigQuery have different sets of custom arguments (for example project-id for BQ versus host for Teradata).
+As you see above, Teradata and BigQuery have different sets of custom arguments (for example project_id for BQ versus host for Teradata).
```
We want to keep the hyphen instead of the underscores in this doc since user can understand the CLI config better. Some folks mistake the underscores and use `--project_id` instead of `--project-id`

nehanene15 reviewed (commented):
Left some minor comments. Could we also add a unit test for the SecretManagerBuilder?

nj1973 commented on data_validation/cli_tools.py:
```diff
@@ -368,16 +368,28 @@ def _configure_validation_config_parser(subparsers):
         "--connection-name", "-c", help="Name of connection used as reference"
     )
+    add_parser.add_argument(
+        "--secret-manger-type",
```
There's a type here. "manger" should be "manager".
This has propagated throughout the changes, if you search for "manger" throughout all changes there are 90 occurrences that are missing an "a".

mokhahmed replied:
fixed the typo to manager instead of manger

mokhahmed committed:
code review fixes
- fix manger typo to manager
- rename argument description
- add google-cloud-secret-manager to requirement file

mokhahmed committed:
code review fixes
- revert the naming of variables connections.md

mokhahmed committed:
code review fixes
- revert the naming of variables connections.md

mokhahmed committed:
code review fixes
- revert the naming of variables connections.md

mokhahmed committed:
code review fixes
- revert the naming of variables connections.md

mokhahmed committed:
Merge branch 'develop' into secret-manager

# Conflicts:
#	setup.py

nehanene15 reviewed (commented):
Logic looks good; we should add unit tests for the secret manager class.

mokhahmed committed:
code review fixes
- add test cases for secret manager class

mokhahmed committed:
code review fixes
- revert the naming of variables connections.md

mokhahmed committed:
code review fixes
- move secret manager testcases

mokhahmed committed:
code review fixes
- revert the naming of variables connections.md
